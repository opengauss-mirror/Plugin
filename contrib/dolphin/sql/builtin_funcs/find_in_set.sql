drop database if exists db_find_in_set;
create database db_find_in_set dbcompatibility 'B';
\c db_find_in_set
select find_in_set(1,'a,1,c');
select find_in_set(1,'true,1,c');
select find_in_set(1.2,'a,1.2,c');
select find_in_set('bac','a,bac,c');
select find_in_set('bac');
select find_in_set('1','a,1,c');
select find_in_set('1.1','a,1.1,c');
select find_in_set('qw','a,1.2,c,qwee,1212,1.1,12,qw');
select find_in_set('qw','a,1.2,null,c,qwee,1212,1.1,12,qw');
select find_in_set(1,'true,1,a,1.2,null,c,qwee,1212,1.1,12,qw');
select find_in_set('测试','a,测试,1.2,c,qwee,1212,1.1');
select find_in_set('1','1,1.2,c,qwee,1212,1.1,12,qw');
select find_in_set(1,'1,1.2,c,qwee,1212,1.1,12,qw');
select find_in_set(1,'1.1,1.2,c,qwee,1212,1.1,12,1');
select find_in_set(1.1,'a,1.2,c,qwee,1212,1.1');
\c postgres
drop database if exists db_find_in_set;
