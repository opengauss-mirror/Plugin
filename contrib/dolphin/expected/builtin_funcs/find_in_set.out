create schema db_find_in_set;
set current_schema to 'db_find_in_set';
select find_in_set(1,'a,1,c');
 find_in_set 
-------------
           2
(1 row)

select find_in_set(1,'true,1,c');
 find_in_set 
-------------
           2
(1 row)

select find_in_set(1.2,'a,1.2,c');
 find_in_set 
-------------
           2
(1 row)

select find_in_set('bac','a,bac,c');
 find_in_set 
-------------
           2
(1 row)

select find_in_set('bac');
ERROR:  function find_in_set(unknown) does not exist
LINE 1: select find_in_set('bac');
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
CONTEXT:  referenced column: find_in_set
select find_in_set('1','a,1,c');
 find_in_set 
-------------
           2
(1 row)

select find_in_set('1.1','a,1.1,c');
 find_in_set 
-------------
           2
(1 row)

select find_in_set('qw','a,1.2,c,qwee,1212,1.1,12,qw');
 find_in_set 
-------------
           8
(1 row)

select find_in_set('qw','a,1.2,null,c,qwee,1212,1.1,12,qw');
 find_in_set 
-------------
           9
(1 row)

select find_in_set(1,'true,1,a,1.2,null,c,qwee,1212,1.1,12,qw');
 find_in_set 
-------------
           2
(1 row)

select find_in_set('测试','a,测试,1.2,c,qwee,1212,1.1');
 find_in_set 
-------------
           2
(1 row)

select find_in_set('1','1,1.2,c,qwee,1212,1.1,12,qw');
 find_in_set 
-------------
           1
(1 row)

select find_in_set(1,'1,1.2,c,qwee,1212,1.1,12,qw');
 find_in_set 
-------------
           1
(1 row)

select find_in_set(1,'1.1,1.2,c,qwee,1212,1.1,12,1');
 find_in_set 
-------------
           8
(1 row)

select find_in_set(1.1,'a,1.2,c,qwee,1212,1.1');
 find_in_set 
-------------
           6
(1 row)

select find_in_set('','a,b,c,,dx');
 find_in_set 
-------------
           4
(1 row)

select find_in_set('','');
 find_in_set 
-------------
           0
(1 row)

select find_in_set('a','a');
 find_in_set 
-------------
           1
(1 row)

select find_in_set(1,'1');
 find_in_set 
-------------
           1
(1 row)

drop schema db_find_in_set cascade;
reset current_schema;
